### ISSUE TITLE
Trigger integrations when submissions are marked as "not spam"

### ISSUE THREAD

**Issue by chrismlusk** [verbb/formie#1724]
### What are you trying to do?

When legitimate submissions are incorrectly marked as spam, we can edit them to mark them as not spam. 

However, any integrations enabled for the form that the submission was made to have not run — which makes sense given that it was thought to be spam. 

But in this scenario, you then have to re-edit each of these submissions and select the integration(s) to run, which can be cumbersome when there are multiple submissions.

### What's your proposed solution?

Would it be possible to automatically trigger integrations when moving a submission from Spam → Not Spam? If not automatically, how about an `ElementAction`?

**Comment by engram-design** [comment]
Yeah, it's come up before a few times for [email](https://github.com/verbb/formie/issues/367) [notifications](https://github.com/verbb/formie/issues/699), I've just not come up with a nice UI to handle this. I **really** don't like the idea of it being automatic, as it then becomes a "magic" thing that happens and the end-user has no idea of the implications behind that.

For example, what if the submission was made before an integration was setup, or before any email notification changes were made? We shouldn't just assume that everything is okay to smash out potentially a whole lot of integrations/email notifications automatically. As I mentioned, you can also bulk set spam/not spam, so there's some potential havok that can be encountered if you don't know what you're going.

So I think we'll need to present users with a choice on how to trigger things when changing a submission (spam or status)

**Comment by chrismlusk** [comment]
Fair points. It is certainly safer to require a deliberate action to re-trigger these things. I guess I'd consider an `ElementAction` to be deliberate, though. 

I suppose you could still end up with potential havoc where someone bulk runs integrations that already ran, which could have undesirable effects (e.g., duplicate leads created in a CRM). But currently that can already happen, there's just more friction in place because you have to edit each submission individually.

Some pie-in-the-sky thoughts off the cuff...

What if integrations could be configured with an optional limit? If submissions tracked the status of their assigned integrations and the number of times they run, that could guard against running things too many times as defined in the form settings.

Of course I'm sure there's a lot of complexity I'm not considering, but I'm happy to kick around ideas here!

**Comment by engram-design** [comment]
Added in [4.0.0-beta.6](https://github.com/verbb/formie/releases/tag/4.0.0-beta.6).
